### ISSUE TITLE
I don't want the label or div wrapper, just the input element.

### ISSUE THREAD

**Issue by ghost** [sir-dunxalot/ember-easy-form-extensions#24]
This line in my template

``` hbs
{{input searchTerm action="search" placeholder="Search..." id="searchThing"}}
```

creates a `<label>` and a `<div>`, and then creates the `<input>` inside the `<div>`.

``` html
<label id="ember1311" class="ember-view control-label">Search term</label>
<div>
   <input id="searchThing" class="ember-view form-control ember-text-field" placeholder="Search..." type="text">
</div>
```

I don't want the `<label>` or the `<div>` elements, just the `<input>`. How do I stop them from being generated?

**Comment by sir-dunxalot** [comment]
You can customize the default easy form templates at `templates/easy-form/whatever.hbs` where `whatever` is the name of the easy form view. In this case, you want to add a file at `app/templates/easy-form/input.hbs` and mirror the content in [input.hbs](https://github.com/sir-dunxalot/ember-easy-form-extensions/blob/master/app/templates/easy-form/input.hbs) without the elements you don't want.

**Comment by ghost** [comment]
Thanks for the quick reply. I was unable to get this to work with either your linked file mirrored exactly or with my edited version of that file.
Can you please explain what this line in your linked [input.hbs](https://github.com/sir-dunxalot/ember-easy-form-extensions/blob/master/app/templates/easy-form/input.hbs) file means?

``` hbs
    {{partial 'easy-form/input-controls'}}
```

If I understand you correctly, the newly-created file `templates/easy-form/input.hbs` would consist of just this line, in order to show only an input element.
Does this line mean that it is looking for a file called `templates/easy-form/input-controls.hbs`?
I do not have such file, nor does the string "input-controls" exist anywhere else in the project.
I am not sure why this is not working, but just thought maybe this had something to do with it.

In summary, I created `templates/easy-form/input.hbs` both exactly as yours was, and with only the above mentioned line. I did not change the template where I am trying to insert the input element.

**Comment by sir-dunxalot** [comment]
Hmm. That's odd. Did you try restarting your Ember CLI server after adding the template? Sometimes the new file isn't picked up until that happens.

The content in `templates/easy-form/input.hbs` almost mirrors the [easy form template](https://github.com/dockyard/ember-easy-form/blob/master/packages/ember-easyForm/lib/templates/input.js) exactly, minor some markup tweaks. The view structure is the same - this addon essentially just makes it compatible with the lastest Ember releases.

The line referenced above will insert the named template and will use the same scope as the surrounding Handlebars. Because it uses the partial helper it will look for the named path inside the `app/templates` directory. You do not need to have the named file there in this case because it is included with this addon. The addon's app file tree is merged with your Ember application's file tree by Ember CLI.

Therefore, your approach was correct. What error were you seeing?

**Comment by ghost** [comment]
It's not an error. It's just acting like I did nothing.
But I just realized that I didn't even have a `templates/easy-form` directory prior to opening this issue. I'm going to look into this. I suspect ember-easy-form-extensions did not get installed correctly. I may have uninstalled it during a previous tinkering.

I'll keep you posted.

EDIT: running `ember install ember-easy-form-extensions` did not create a `templates/easy-form` directory. Should it have?

**Comment by sir-dunxalot** [comment]
You won't have the directory there. Ember CLI uses a tool called Broccoli to build your application. An addon is a 'tree' of files and your app is a 'tree' of files. Both this addon and your app have an `app` directory. The files and folders in that directory (aka the app trees) get merged together and are compiled into a single tree (or directory) and end up as a single javascript file called `app.js` - so you won't ever see addon files 'installed' into the app unless you look in your compiled javascript after running `ember s` or `ember build`. The files in your app's app tree overwrite the files in this addon's app tree when two files are at the same path.

Thus, by adding a template in your application at `templates/easy-form/input.hbs` you will overwrite the file this addon would have used (i.e. the file in this addon's app tree at  `ember-easy-form-extensions/app/templates/easy-form/input.hbs`). Does that make sense?

So you only have to add the one file in the new `templates/easy-form` directory and it should work. I'm not sure why it wouldn't be finding your template. Did you try the server restart?

**Comment by ghost** [comment]
I did try the server restart. I will review you detailed message (thanks for that), do some research, and rebuild. I'll post results.

**Comment by ghost** [comment]
Is there a way to turn off easy-form (and extensions) for certain forms in my app?

**Comment by sir-dunxalot** [comment]
Yep - if you use the ember input syntax it will not add an easy form input. For example, instead of:

``` hbs
{{input description as='text'}}
```

You can do:

``` hbs
{{input value=description type='text'}}
```

This will use an Ember input instead of an easy form input. As for the easy form extensions stuff you can use `<form>` instead of `{{form-wrapper}}` and so forth - just make sure to handle the `submit` event in your view using `<button type="submit">` inside the form (or similar).

**Comment by ghost** [comment]
My code originally looked like this:

``` hbs
{{input value=description type='text'}}
```

and the problem was that the actual string "description" started out in the input element, as if the code said `value="description"`

Any idea what would cause this?

This came up after we upgraded ember (1.8 to 1.11.1), ember-cli (0.1.5 to 0.2.3), and ember data (to latest). During this process, ember-easy-form-extensions was installed, to get ember-easy-forms to work with htmlbars.

**Comment by sir-dunxalot** [comment]
Actually, yeah. Ember Easy Form seems to hack around the input. If you do `valueBinding='description'` it will probably work. Unfortunately, this throws a deprecation warning in the latest Ember release.

I've got this addon working with Ember 1.10 but there are some known issues with 1.11 due to the Ember API changes. Knowing what the future holds with Ember releases, I'm considering a rewrite - providing the Easy Form API but with very different code until Easy Form is rewritten using scoped helpers (a future Ember feature).

**Comment by sir-dunxalot** [comment]
I'll just keep this issue open for now.

**Comment by ghost** [comment]
Your `valueBinding='description'` suggestion worked. But, I have yet to look deeper to make sure everything is working after the upgrade. Do you suggest rolling back to Ember 1.10 until you finish your rewrite? Do the "some know issues with 1.11 due to the Ember API changes" break things, or just cause deprecation warnings?

I will look deeper now, to see if anything else is broke.

**Comment by sir-dunxalot** [comment]
I wouldn't recommend staying at 1.10 just for this library, unless forms and user-friendly validations are a huge part of your app. The issues re: 1.11 are a mix of errors and warnings.

The biggest problem here is the lack of progress from [easy form](https://github.com/dockyard/ember-easy-form). I will basically rewrite the whole thing to make it work with 1.11.

**Comment by ghost** [comment]
I'll just deal with the deprecation warnings until you release your rewrite.

Based on my receiving no replies when I open easy-form issues, I think your best hope is to remove reliance on easy form and bring it all under your umbrella. I might be too new to ember to provide much help, but I can assist in any way I can, maybe in testing.  Someday I hope to write my own addons, so this would be good experience. Just message me if you need anything, and I'll keep an eye out for your rewrite.

**Comment by sir-dunxalot** [comment]
Thanks Jerry, I appreciate it! I'll definitely let you know.

I think the best course of action will be for me to rewrite Easy Form using the latest Ember APIs (but match the Easy Form API) and then I'll make a point of adding improvements and issues that are listed as opportunities to jump into the code. Testing would be a great area for that.

**Comment by ghost** [comment]
That all sounds great. I'm watching this project, but feel free to reach out directly anytime.

**Comment by ghost** [comment]
Do you know a fix similar to the valueBinding fix to use for checkboxes? Mine are always checked.

``` hbs
{{input type="checkbox" checked=activity.selected}}
```

This always has the checkbox checked upon initial page load. Have you seen ember-easy-form cause something like this before?

**Comment by sir-dunxalot** [comment]
Did you try `checkedBinding`? I wrote a custom checkbox for personal apps so haven't run into this problem.

Also, I already started work on a Ember CLI-ified branch at 67ece81a32117dce73e8a6e9f6aa9800fd0a6115, which will solve a lot of the issues with the binding.

**Comment by ghost** [comment]
The `checkedBinding` worked. Thanks for that. I knew it was something like that.

Great news about the branch. I'll check it out.

**Comment by sir-dunxalot** [comment]
The new branch now has it's first prerelease for testing and feedback. [You can test the prerelease here](https://github.com/sir-dunxalot/ember-easy-form-extensions/releases/tag/1.0.0.beta). Please note the API for inputs has changed slightly from the original Easy Form syntax.
